### DISCUSSION TITLE
Chip/Pill Input

### DISCUSSION THREAD

**Discussion by bthomasstill** [twilio-labs/paste#3311] [Ideas] [answered]
Are there any plans to make a chip input style component? We have a usecase for something like in this video --

https://github.com/twilio-labs/paste/assets/114398889/a6907893-a31f-4285-bdb7-f4d75113bbdd

I've seen similar discussion around chips/pills to display selected options within the combobox but nothing similar for this usecase where it would just be an input. I thought it might be possible replicate the UI with a TextArea and FormPill but it seems TextArea doesn't play nicely having components inside it, since it's expecting only text?

**Comment by SiTaggart** [comment, marked as answer]
Hi @bthomasstill 

We've not really considered this as part of the roadmap officially. We feel it's fairly straight forward to compose with the pieces we provide, it might just not be as obvious as we'd like it to be.

I knocked up a quick code sandbox to demonstrate: https://codesandbox.io/s/composer-and-pill-custom-mc56h3?file=/src/index.tsx

The first part of it is acknowledging you can't place anything in HTML input elements other than strings. Essentially, even with the multi select combobox, the pills are never in the input. They just look like they are. They are siblings to the input with a border placed around them both. Internally, we use `InputBox` and `InputElement` to achieve this. Element being "the thing you type in" and Box being "the things that looks like a box". It allows us to "place" things inside an input but not, actually.

Hope that helps
